From 1d682c6aa68faedb20c283be9e5eb31521e81b0d Mon Sep 17 00:00:00 2001
From: Nikolai Kondrashov <Nikolai.Kondrashov@redhat.com>
Date: Thu, 29 Jan 2015 13:51:29 +0200
Subject: [PATCH 1/1] Return positive integers from each radius_xlat

Make all versions of radius_xlat return positive integers only,
including stubs, to match the result checking.

This fixes the following Coverity errors:

Error: NEGATIVE_RETURNS (CWE-394):
freeradius-server-2.2.6/src/modules/rlm_expr/rlm_expr.c:625: negative_return_fn: Function "radius_xlat(buffer, 1024, fmt, request, func)" returns a negative number.
freeradius-server-2.2.6/src/main/radconf2xml.c:52:2: return_negative_constant: Explicitly returning negative value "-1".
freeradius-server-2.2.6/src/modules/rlm_expr/rlm_expr.c:625: var_assign: Assigning: unsigned variable "len" = "radius_xlat".
freeradius-server-2.2.6/src/modules/rlm_expr/rlm_expr.c:637: negative_returns: "len" is passed to a parameter that cannot be negative.
freeradius-server-2.2.6/src/lib/base64.c:66:50: sizet: "inlen" is a size_t parameter.

Error: NEGATIVE_RETURNS (CWE-394):
freeradius-server-2.2.6/src/modules/rlm_expr/rlm_expr.c:659: negative_return_fn: Function "radius_xlat(buffer, 1024, fmt, request, func)" returns a negative number.
freeradius-server-2.2.6/src/main/radconf2xml.c:52:2: return_negative_constant: Explicitly returning negative value "-1".
freeradius-server-2.2.6/src/modules/rlm_expr/rlm_expr.c:659: var_assign: Assigning: unsigned variable "len" = "radius_xlat".
freeradius-server-2.2.6/src/modules/rlm_expr/rlm_expr.c:666: negative_returns: "len" is passed to a parameter that cannot be negative.
freeradius-server-2.2.6/src/lib/base64.c:315:3: parm_loop_bound: Using unsigned parameter "inlen" in a loop exit test.
---
 src/main/radconf2xml.c | 2 +-
 src/main/radmin.c      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main/radconf2xml.c b/src/main/radconf2xml.c
index 718fd99..afb696c 100644
--- a/src/main/radconf2xml.c
+++ b/src/main/radconf2xml.c
@@ -49,7 +49,7 @@ char *debug_log_file = NULL;
 int radius_xlat(UNUSED char *out, UNUSED int outlen, UNUSED const char *fmt,
 		UNUSED REQUEST *request, UNUSED RADIUS_ESCAPE_STRING func)
 {
-	return -1;
+	return 0;
 }
 
 static int usage(void)
diff --git a/src/main/radmin.c b/src/main/radmin.c
index 35ab715..e648032 100644
--- a/src/main/radmin.c
+++ b/src/main/radmin.c
@@ -93,7 +93,7 @@ char *debug_log_file = NULL;
 int radius_xlat(UNUSED char *out, UNUSED int outlen, UNUSED const char *fmt,
 		UNUSED REQUEST *request, UNUSED RADIUS_ESCAPE_STRING func)
 {
-	return -1;
+	return 0;
 }
 
 static FILE *outputfp = NULL;
-- 
2.1.4

